### PR TITLE
Add `_legacy_feature_flag` for customer threads routes

### LINKED ISSUE #31377: BO - The customer service view page is displayed as a migrated page while the Customer threads feature flag is deactivated
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

Since 8.0.0 we use feature flags to hide migrated pages that are not finished.

The "Customer Service" **View** back-office page migrated to Symfony is accessible when the "Customer threads" feature flag is deactivated. It should not be accessible.

**8.1.0_betabuild1 : NOK** :x: 

image

https://user-images.githubusercontent.com/92912932/219036085-86e79ba6-00c5-49fe-8077-3a8a6201258e.mp4

### Expected behavior

When clicking on the "View" link on the _legacy_ back-office "Customer service" page, we should not be redirected on the _migrated_ "customer service" View back-office page. We should be redirected to legacy Customer service View page instead.

### Steps to reproduce

1. Install the 8.1.0_beta_build1 (branch `8.1.x`)
2. Go to back-office in Advanced parameters > New & Experimental Features. Then make sure the Feature Flag of Customer Threads is _disabled_
3. Go to Back-office SELL > Customers > "Add a new Customer" page, then create a new customer
4. Go to Front-office. You are going to create a fake customer support request 😁 using the customer you just created. Login using the customer login/password then in your "My account" space, go to "Contact US" > Fill in the information of the Customer request with the created customer and submit it.
5. Go to back-office again. Go to SELL > Customer Service > Click on to View: you can see the sent customer request you just created. 
6. Click on "View": you have been redirected to the "View Customer service" Symfony page. It is the problem. You should have been redirected to the "View customer service" legacy page.

### PrestaShop version(s) where the bug happened

8.1.0_beta_build1, develop

### PHP version(s) where the bug happened

7.4

### If your bug is related to a module, specify its name and its version

_No response_

### PR DESCRIPTION
**Pull request by boherm**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 8.1.x
| Description?      | Fix redirect to new symfony controller for customer threads if `customer_threads` flag is disabled.
| Type?             | bug fix
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | See #31377 
| Fixed ticket?     | Fixes #31377 
| Related PRs       | 
| Sponsor company   |

### CODE REVIEW

**Review by 0x346e3730 — DISMISSED** [review]

**Review by matks — DISMISSED** [review]

**Review by HanaRebaiQA — COMMENTED** [review]
Hello @boherm 

I have checked your PR. I found this big issue. All FO is broken with this exception

image

https://user-images.githubusercontent.com/92912932/224974587-a406f475-9fc3-4cba-b386-324993549c3b.mp4


Could you please check it please ? 

Thank You!

**Comment by FabienPapet** [comment]
@HanaRebaiQA you forgot to run `make assets`

**Comment by HanaRebaiQA** [comment]
Hello @FabienPapet 

Sorry, but I did with `make install `as usual


https://user-images.githubusercontent.com/92912932/224985479-e126bb62-dffc-4581-8348-0debe4a2beb0.mp4

**Comment by ghost** [comment]
Only `make` ?

**Comment by FabienPapet** [comment]
I don't know what the real reason is, but is it related on your local environment, and does not come from the PR.

**Review by HanaRebaiQA — COMMENTED** [review]
Hello again, 

Thank you @okom3pom I have re-installed again using only the `make` and the FO is well displayed!

@boherm I found these issues while testing your PR : 

**Issue 1 :** The customer View page is displayed as a migrated page when activating the Customer threads feature flag and then deactivated it and refresh the page 

1. Go to FO > Add a new customer service 
2. Click on Send
3. Go to BO > CONFIGURE > Advanced parameters > New & Experimental Features > **Disable** the FF of Customer threads
4. Go to BO > SELL > Customer Service > Click to view the created customer service
5. See that the page is displayed on legacy 
6. Come back to BO > CONFIGURE > Advanced parameters > New & Experimental Features > **Enable** the FF of Customer threads
7. Come back to the View customer Service page > refresh the page 
8. See that the page is displayed on a symphony
9.  Come back to BO > CONFIGURE > Advanced parameters > New & Experimental Features > **Disable** the FF of Customer threads
10. Come back to the View customer Service page > refresh the page 
11. See error > the page is displayed always on a symphony !! 
12. When coming back to the Customer service page and clicking again on View > The page is on legacy OK


https://user-images.githubusercontent.com/92912932/225034720-159b3309-4333-4c10-ad52-07c5d59eaa24.mp4

image

image

**PS** : When the View customer page is displayed with the legacy page and then enabling the FF of the Customer threads and go back to refresh the view customer page ==> The page is changed to migrated : OK 



**Issue 2:** Related to automated tests (eg: Customer service is failed)

Compared to Nightly, only the API failed on 8.1.x 
**Nightly :** https://github.com/PrestaShop/PrestaShop/actions/runs/4411247942
**Run tests :** https://github.com/HanaRebaiQA/testing_pr/actions/runs/4414273996/jobs/7740868411

**PS :** I have launched the failed jobs tests many times
ping @PrestaShop/qa-automation What do you think? Are there failures related to this PR ?

@boherm Could you please check it ? 

Thank You!

**Comment by boherm** [comment]
Thanks @HanaRebaiQA ! I will check that :)

**Comment by ghost** [comment]
@HanaRebaiQA before 1.7.7x you having to do `make install` and `make assets`, now you can use only `make`.

These commands depend on what you need to do but for tests using only `make` will take more time but you are sure.

@FabienPapet also gives the right answer with `make assets`

I will look to add it in the docs if it is not the case

**Comment by boherm** [comment]
@HanaRebaiQA For the first issue in my PR, this is a normal behaviour. 
If we have this feature flag enabled, legacy urls are convert into new one and legacy page are redirected to new one too.
But, the reverse isn't performed by routing and not only on my PR.

For the second issue with automated tests on UI, after check with @Progi1984 and @jolelievre we found that tests `functional:BO:customer-service` are based on symfony template and not legacy one! It was introduced by https://github.com/PrestaShop/PrestaShop/pull/29794.
@Progi1984 will check soon to make tests pass again for Customer Threads!

**Comment by Progi1984** [comment]
@boherm I fixed UI tests (and rebased it).

**Review by jolelievre — DISMISSED** [review]
Thanks @boherm and @Progi1984

**Review by nesrineabdmouleh — DISMISSED** [review]

**Review by djoelleuch — COMMENTED** [review]
Hello @boherm ,

 I Compared to Nightly on 8.1.x , it must all passed  ,but it 's not the case .

image

Nightly : https://github.com/PrestaShop/PrestaShop/actions/runs/4432610252
Run tests : https://github.com/djoelleuch/ga.tests.ui.pr/actions/runs/4435209357

PS : I have launched the failed jobs tests many times
ping @PrestaShop/qa-automation what do you think ? 

@boherm  Could you please check it ?

Thank You!

**Comment by Progi1984** [comment]
@djoelleuch & @boherm I will check it tomorrow ;) (I'm on the :train:).

File: tests/UI/campaigns/functional/FO/05_contactUs/04_addAttachment.ts [hunk 1]
```diff
@@ -174,10 +174,11 @@ describe('FO - Contact us : Add attachment', async () => {
     await expect(badgeNumber).to.contains(idCustomer);
 
     const text = await viewPage.getCustomerMessage(page);
+    console.log(text);
```
Comment (nesrineabdmouleh) [inline tests/UI/campaigns/functional/FO/05_contactUs/04_addAttachment.ts]:
```suggestion
```

Comment (Progi1984) [reply tests/UI/campaigns/functional/FO/05_contactUs/04_addAttachment.ts]: Fixed in d924abd. Thanks

**Comment by Progi1984** [comment]
@djoelleuch  The CI is now :green_circle: : https://github.com/Progi1984/ga.tests.ui.pr/actions/runs/4446279618. Since your tests, only UI tests have been fixed. So I think it's OK for "QA :heavy_check_mark:"

**Review by nesrineabdmouleh — APPROVED** [review]

**Review by djoelleuch — APPROVED** [review]
HI @boherm ,
LGTM , the automated tests are OK.
It's QA :heavy_check_mark: 
thank you !

**Comment by jolelievre** [comment]
Thanks @boherm and @djoelleuch

**Review by jolelievre — APPROVED** [review]

**Pull request merged**